lib/task/digitalobject/digitalObjectRegenDerivativesTask.class.php

```diff
@@ -174,9 +174,25 @@ protected function execute($arguments = array(), $options = array())
 
   public static function regenerateDerivatives(&$digitalObject, $type = null)
   {
+    // Determine usage ID for type
+    switch($type)
+    {
+      case "reference":
+        $usageId = QubitTerm::REFERENCE_ID;
+        break;
+
+      case "thumbnail":
+        $usageId = QubitTerm::THUMBNAIL_ID;
+        break;
+
+      default:
+        $usageId = QubitTerm::MASTER_ID;
+    }
+
     // Delete existing derivatives
     $criteria = new Criteria;
     $criteria->add(QubitDigitalObject::PARENT_ID, $digitalObject->id);
+    $criteria->add(QubitDigitalObject::USAGE_ID, $usageId);
```

Comment:
Hmm, there should never be a case where `$usageId == QubitTerm::MASTER_ID` and `parent_id <> NULL`.  In practice this means the that when `$usageId = QubitTerm::MASTER_ID` this query will return nothing, and I think it should return all derivatives.  You could wrap this line in an `if (QubitTerm::MASTER_ID != $usageId)` conditional to produce the desired results.